src/mca/pdl/pdlopen/pdl_pdlopen_module.c

```diff
@@ -175,7 +175,26 @@ static int pdlopen_foreachfile(const char *search_path,
 
         dp = opendir(dirs[i]);
         if (NULL == dp) {
-            ret = PMIX_ERR_IN_ERRNO;
+            if (0 == i) {
```

Comment:
This is in keeping with what is in OPAL. If x can't be opened, we immediately return and never look beyond that point. All I'm doing is emitting the warning if that happens with the very first entry as this would correspond with the typical use case of PMIX_INSTALL_PREFIX being incorrectly set. Rather than doing something different here, I'd suggest taking that question upstream to the OPAL community - there was a reason behind the original logic.